src/terms/local.jl

```diff
@@ -90,8 +90,9 @@ function (::AtomicLocal)(basis::PlaneWaveBasis{T}) where {T}
 end
 
 @timing "forces: local" function compute_forces(::TermAtomicLocal,
-                                                basis::PlaneWaveBasis{T},
-                                                ψ, occ; ρ, kwargs...) where {T}
+                                                basis::PlaneWaveBasis,
+                                                ψ, occ; ρ, kwargs...)
+    T = real(eltype(ψ[1]))
     atoms = basis.model.atoms
```

Comment:
we've already had this issue I think, and I'm not sure where we landed. Should we promote T from the eltype of basis and that of T? CC @niklasschmitz

Reply:
Yup, I remember we talked together about that in Roscoff, but then I kept the modification in my branch :sweat_smile:

Reply:
`promote_type` sounds fair to me. The only potential issue I see is that it can introduce very subtle type instabilities if we are not careful (e.g. `promote_type(ForwardDiff.Dual{Float64}, Float64) == Real` only `promote_type(ForwardDiff.Dual{Float64, 1, 1}, Float64) <: Dual`).

Reply:
ok but with forwarddiff all types should be the latter, not the former, no?

Reply:
sure. I just wanted to point out that in case the input is garbage the `promote_type` approach will make the function still work and in this way effectively hide the problem. That could make debugging harder in the future.

Reply:
So I should use `promote_type(eltype(basis), real(eltype(ψ[1])))` ?

Reply:
Exactly.